frontend/public/module/k8s/k8s-models.ts

```diff
@@ -27,7 +27,7 @@ export const modelsToMap = (models: K8sKind[]): ImmutableMap<K8sResourceKindRefe
 let k8sModels = modelsToMap(_.values(staticModels));
 
 const hasModel = (model: K8sKind) =>
-  k8sModels.has(referenceForModel(model)) || k8sModels.has(model.kind);
+  k8sModels.has(referenceForModel(model)) || (!model.crd && k8sModels.has(model.kind));
```

Comment:
Now that you got it to work where I failed... I would recommend going with my first implementation and making `hasModel` and `modelsToMap` use the same logic to figure out the key to store it under.

From Slack:

> https://github.com/spadgett/console/blob/586f68d0eaa32defc34df2554b39c5fc226a7d79/frontend/public/module/k8s/k8s-models.ts#L13-L18
> I think we need to de-duplicate and reuse this logic. The hasModel should lookup in the same order/reference as the storage.

Reply:
Suggested change:

```
const modelKey = (model) => {
  // TODO: Use `referenceForModel` even for known API objects
  return model.crd ? referenceForModel(model) : model.kind;
};

export const modelsToMap = (models: K8sKind[]): ImmutableMap<K8sResourceKindReference, K8sKind> => {
  return ImmutableMap<K8sResourceKindReference, K8sKind>().withMutations((map) => {
    models.forEach((model) => map.set(modelKey(model), model));
  });
};
...
const hasModel = (model: K8sKind) => k8sModels.has(modelKey(model));
```

That way they are both based off the same value, it looks it up in the same fashion it stores it. No way for it to be confused on reverse lookups.

Reply:
Unfortunately, this changes the logic a bit and breaks when you pass the full reference for a model that has `crd: false`. For instance, `route.openshift.io~v1~Route` will no longer locate the route model.

Reply:
Disregard, you're right